File: src/Analysim.Web/ClientApp/src/app/interfaces/project-comment-flag.ts
export interface ProjectCommentFlag {
  flagID: number;
  commentID: number;
  userID: number;
  commentContentSnapshot: string;
  createdAt: string;
}

export interface FlaggedCommentGroup {
  commentID: number;
  commentOwnerUsername: string;
  commentProjectName: string;
  commentProjectOwner: string;
  flags: ProjectCommentFlagRow[];
}

export interface ProjectCommentFlagRow {
  flagID: number;
  commentContentSnapshot: string;
  flaggedByUsername: string;
  createdAt: string;
}

File: src/Analysim.Core/Entities/ProjectCommentFlag.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Core.Entities
{
    public class ProjectCommentFlag
    {
        // PK
        [KeyAttribute]
        public int FlagID { get; set; }

        // Comment
        [ForeignKey("ProjectComment")]
        public int CommentID { get; set; }
        public ProjectComment ProjectComment { get; set; } = null!;

        // Flagger
        [ForeignKey("User")]
        public int UserID { get; set; }
        public User User { get; set; } = null!;

        // Snapshot comment content
        public string CommentContentSnapshot { get; set; } = null!;

        // Timestamp
        public DateTime CreatedAt { get; set; }
    }
}